Libraries/LibWeb/HTML/Navigable.cpp

```diff
@@ -1315,15 +1316,18 @@ WebIDL::ExceptionOr<void> Navigable::populate_session_history_entry_document(
         return {};
 
     // 5. Queue a global task on the navigation and traversal task source, given navigable's active window, to run these steps:
-    queue_global_task(Task::Source::NavigationAndTraversal, *active_window(), GC::create_function(heap(), [this, entry, navigation_params = move(navigation_params), navigation_id, user_involvement, completion_steps, csp_navigation_type]() mutable {
+    queue_global_task(Task::Source::NavigationAndTraversal, *active_window(), GC::create_function(heap(), [this, entry, promise, navigation_params = move(navigation_params), navigation_id, user_involvement, completion_steps, csp_navigation_type]() mutable {
         // NOTE: This check is not in the spec but we should not continue navigation if navigable has been destroyed.
         if (has_been_destroyed())
             return;
 
         // 1. If navigable's ongoing navigation no longer equals navigationId, then run completionSteps and abort these steps.
         if (navigation_id.has_value() && (!ongoing_navigation().has<String>() || ongoing_navigation().get<String>() != *navigation_id)) {
-            if (completion_steps)
+            if (completion_steps) {
+                // Note: Use Core::Promise to signal SessionHistoryTraversalQueue that it can continue to execute next entry.
+                promise->resolve({});
```

Comment:
it’d probably be good to name the `Promise` variables differently than plain `promise` (everywhere), e.g. `continue_session_history_processing_signal` or something along those lines?